scanpointgenerator/generators/staticpointgenerator.py

```diff
@@ -11,16 +11,32 @@
 #
 ###
 
-from scanpointgenerator.core import Generator, ASize
+import numpy as np
+from scanpointgenerator.core import Generator, ASize, UAxes, AAxes
 
 
 @Generator.register_subclass(
     "scanpointgenerator:generator/StaticPointGenerator:1.0")
 class StaticPointGenerator(Generator):
-    """Generate 'empty' points with no axis information"""
-    def __init__(self, size):
-        # type: (ASize) -> None
-        super(StaticPointGenerator, self).__init__(axes=[], units=[], size=size)
+    """Generate 'empty' points with optional axis information"""
+    def __init__(self, size, axes=[]):
+        # type: (ASize, UAxes) -> None
+
+        axis = AAxes(axes)
+        # Validate
+        assert len(axis) <= 1, \
+            "Expected 1D or no axes, got axes %s" % (
+                list(axis))
+
+        super(StaticPointGenerator, self).__init__(axes=axis,
+                                                   units=[""] * len(axis),
+                                                   size=size)
 
     def prepare_arrays(self, index_array):
-        return {}
+        arrays = {}
+        # If there is an axis in self.axes, produce a 1-indexed array for it
+        for axis in self.axes:
+            arrays[axis] = np.arange(1, len(index_array)+1)
```

Comment:
Actually, this is just index_array + 1

Reply:
Even though the index_array sometimes doesn't have values 0, 1, 2, 3 .... ?
e.g. in one unit tests, index_array has the value [-0.5, 0.5, 1.5, 2.5, 3.5, 4.5]